### DISCUSSION TITLE
How to handle versioning in path (e.g., "/api/v1/...")

### DISCUSSION THREAD

**Discussion by joergrech** [jellydn/next-swagger-doc#846] [Q&A] [answered]
Hi,
  I just started a version 2 of my API and while next-swagger-doc generates the docs as intended, it combines all endpoints from v1 and v2. Is ist possible to separate the versions or hide the v1 endpoints?

My paths:
```
/api/v1/...
/api/v2/...
```

I want to show the docs separately (e.g., with a selector on the page for every version or via different paths) or only show the docs from the v2 subdirectory.

**Comment by jellydn** [comment, marked as answer]
Hi @joergrech I think you could use `apiFolder` option for your need.


```typescript
import { createSwaggerSpec } from https://github.com/jellydn/next-swagger-doc;

export const getApiV1Docs = async () => {
  const spec = createSwaggerSpec({
    apiFolder: 'api/v1', // define api folder
    definition: {
      openapi: '3.0.0',
      info: {
        title: 'Next Swagger API Example',
        version: '1.0',
      },
      components: {
        securitySchemes: {
          BearerAuth: {
            type: 'http',
            scheme: 'bearer',
            bearerFormat: 'JWT',
          },
        },
      },
      security: [],
    },
  });
  return spec;
};
```

Let me know if any issue.

**Reply by joergrech** [reply]
Thanks for the tipp! It worked with the following line:
```
apiFolder: '/pages/api/v2', // define api folder
```
